### ISSUE TITLE
GetAngleBetweenVectors returning NaN

### ISSUE THREAD

**Issue by Eole211** [BabylonJS/Babylon.js#9611]
Hello there. 
Here's a little glitch that can be pretty annoying sometimes : 
If you give `Vector3.GetAngleBetweenVectors` two vectors with close enough directions (or opposite directions), the function returns `NaN` : 
Example : 
```js
    const a = new BABYLON.Vector3(0.45230270427355385, 0, 0.8918644872999654);
    const a2 = new BABYLON.Vector3(0.45230270427355385, 0, 0.8918644872999653);
    const b = new BABYLON.Vector3(-0.45230270427355385,0, -0.8918644872999653);
   console.log( "similar vectors angle : ", BABYLON.Vector3.GetAngleBetweenVectors(a, a2, BABYLON.Vector3.Up()));
   console.log( "opposite vectors angle : ", BABYLON.Vector3.GetAngleBetweenVectors(a, b, BABYLON.Vector3.Up()));
```
Example in playground : 
https://www.babylonjs-playground.com/#0DASQS#1

**Comment by Popov72** [comment]
Fixed by #9613

**Comment by Eole211** [comment]
nice fix, thanks @Popov72
